File: 백준/Silver/2847. 게임을 만든 동준이/게임을 만든 동준이.py
n = int(input())
score_list = [int(input()) for _ in range(n)]
cnt = 0

for i in range(n - 1, 0, -1):
  if score_list[i] <= score_list[i - 1]:
    minus_value = score_list[i - 1] - score_list[i] + 1 # 이전 값이랑 비교해서 1 큰거
    cnt += minus_value
    score_list[i - 1] = score_list[i] - 1 # 이전 값 변경
    
print(cnt)

File: 백준/Silver/2847. 게임을 만든 동준이/게임을 만든 동준이.js
const fs = require('fs');

function solve(scores) {
  let count = 0;

  for (let i = n - 1; i > 0; i--) {
    if (scores[i] <= scores[i - 1]) {
      let minus_value = scores[i - 1] - scores[i] + 1;
      count += minus_value;
      scores[i - 1] = scores[i] - 1;
    }
  }

  return count;
}

const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');
const n = parseInt(input[0]);
const scores = input.slice(1).map(Number);

const result = solve(scores);
console.log(result);